### PR TITLE
[wip] New c++ domain generator

### PR DESCRIPTION
**Pull request by dawagner**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48021954%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48022071%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48022117%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48022159%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48022223%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48022312%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48022349%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48042716%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48042874%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48043053%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48043204%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48604862%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48604870%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48604931%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48604936%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48605001%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48605004%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48605069%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48605083%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%200b584463582fc436cd0ce0f10a6f4610771cfa7c%20tools/xmlGenerator/domainGeneratorConnector.cpp%2045%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48022349%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Reads%22%2C%20%22created_at%22%3A%20%222015-12-18T13%3A00%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/xmlGenerator/domainGeneratorConnector.cpp%3AL1-205%22%7D%2C%20%22Pull%200b584463582fc436cd0ce0f10a6f4610771cfa7c%20tools/xmlGenerator/domainGenerator.py%20248%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48043204%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Forward%20the%20connector%27s%20return%20code.%22%2C%20%22created_at%22%3A%20%222015-12-18T16%3A35%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-12-30T14%3A04%3A09Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/xmlGenerator/domainGenerator.py%3AL159-208%22%7D%2C%20%22Pull%200b584463582fc436cd0ce0f10a6f4610771cfa7c%20tools/xmlGenerator/domainGeneratorConnector.cpp%20138%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48043053%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Throw%20an%20error.%22%2C%20%22created_at%22%3A%20%222015-12-18T16%3A33%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20The%20code%20now%20continues%20on%20error%20but%20keep%20track%20of%20how%20many%20occurred%20and%20report%20it%20with%20the%20exit%20code.%22%2C%20%22created_at%22%3A%20%222015-12-30T14%3A04%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/xmlGenerator/domainGeneratorConnector.cpp%3AL1-205%22%7D%2C%20%22Pull%200b584463582fc436cd0ce0f10a6f4610771cfa7c%20tools/xmlGenerator/domainGeneratorConnector.cpp%2097%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48042874%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Missing%20closing%20%60%27%60%20before%20%60%20rejected%60.%22%2C%20%22created_at%22%3A%20%222015-12-18T16%3A32%3A09Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/xmlGenerator/domainGeneratorConnector.cpp%3AL1-205%22%7D%2C%20%22Pull%200b584463582fc436cd0ce0f10a6f4610771cfa7c%20tools/xmlGenerator/domainGenerator.py%20246%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48022117%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22flush%20may%20not%20be%20needed.%22%2C%20%22created_at%22%3A%20%222015-12-18T12%3A56%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-12-30T14%3A00%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/xmlGenerator/domainGenerator.py%3AL159-208%22%7D%2C%20%22Pull%200b584463582fc436cd0ce0f10a6f4610771cfa7c%20tools/xmlGenerator/domainGeneratorConnector.cpp%2090%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48042716%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%60numericalValue%60%20%28for%20naming%20consistency%29%22%2C%20%22created_at%22%3A%20%222015-12-18T16%3A30%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/xmlGenerator/domainGeneratorConnector.cpp%3AL1-205%22%7D%2C%20%22Pull%200b584463582fc436cd0ce0f10a6f4610771cfa7c%20newDomainGenerator.py%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48021954%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22oops%2C%20I%20did%20not%20intend%20to%20add%20this%20file.%20I%27ll%20remove%20it.%22%2C%20%22created_at%22%3A%20%222015-12-18T12%3A54%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-12-30T14%3A05%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20newDomainGenerator.py%3AL1-324%22%7D%2C%20%22Pull%200b584463582fc436cd0ce0f10a6f4610771cfa7c%20tools/xmlGenerator/domainGeneratorConnector.cpp%2011%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48022223%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Add%20a%20custom%20exception%20class%20to%20differentiate%20Parameter%20Framework%20errors%20from%20other%20ones.%22%2C%20%22created_at%22%3A%20%222015-12-18T12%3A58%3A09Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/xmlGenerator/domainGeneratorConnector.cpp%3AL1-205%22%7D%2C%20%22Pull%200b584463582fc436cd0ce0f10a6f4610771cfa7c%20tools/xmlGenerator/domainGeneratorConnector.cpp%2029%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48022312%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Maybe%20we%20could%20unconditionally%20instantiate%20the%20logger%3A%20it%20doesn%27t%20cost%20much.%22%2C%20%22created_at%22%3A%20%222015-12-18T12%3A59%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20think%20it%20is%20just%20as%20good%20as%20is.%20No%20need%20to%20complexity%20the%20code.%22%2C%20%22created_at%22%3A%20%222015-12-30T14%3A02%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-12-30T14%3A02%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/xmlGenerator/domainGeneratorConnector.cpp%3AL1-205%22%7D%2C%20%22Pull%200b584463582fc436cd0ce0f10a6f4610771cfa7c%20tools/xmlGenerator/domainGenerator.py%20223%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48022071%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22no%20need%20to%20flush%20here.%22%2C%20%22created_at%22%3A%20%222015-12-18T12%3A56%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22done%22%2C%20%22created_at%22%3A%20%222015-12-30T14%3A00%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-12-30T14%3A05%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/xmlGenerator/domainGenerator.py%3AL159-208%22%7D%2C%20%22Pull%200b584463582fc436cd0ce0f10a6f4610771cfa7c%20tools/xmlGenerator/domainGeneratorConnector.cpp%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/322%23discussion_r48022159%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22missing%20license%20header%22%2C%20%22created_at%22%3A%20%222015-12-18T12%3A57%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20tools/xmlGenerator/domainGeneratorConnector.cpp%3AL1-205%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 0b584463582fc436cd0ce0f10a6f4610771cfa7c tools/xmlGenerator/domainGeneratorConnector.cpp 1'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/322#discussion_r48022159'>File: tools/xmlGenerator/domainGeneratorConnector.cpp:L1-205</a></b>
- <a href='https://github.com/dawagner'></a> missing license header
- [ ] <a href='#crh-comment-Pull 0b584463582fc436cd0ce0f10a6f4610771cfa7c tools/xmlGenerator/domainGeneratorConnector.cpp 11'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/322#discussion_r48022223'>File: tools/xmlGenerator/domainGeneratorConnector.cpp:L1-205</a></b>
- <a href='https://github.com/dawagner'></a> Add a custom exception class to differentiate Parameter Framework errors from other ones.
- [ ] <a href='#crh-comment-Pull 0b584463582fc436cd0ce0f10a6f4610771cfa7c tools/xmlGenerator/domainGeneratorConnector.cpp 45'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/322#discussion_r48022349'>File: tools/xmlGenerator/domainGeneratorConnector.cpp:L1-205</a></b>
- <a href='https://github.com/dawagner'></a> Reads
- [ ] <a href='#crh-comment-Pull 0b584463582fc436cd0ce0f10a6f4610771cfa7c tools/xmlGenerator/domainGeneratorConnector.cpp 90'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/322#discussion_r48042716'>File: tools/xmlGenerator/domainGeneratorConnector.cpp:L1-205</a></b>
- <a href='https://github.com/dawagner'></a> `numericalValue` (for naming consistency)
- [ ] <a href='#crh-comment-Pull 0b584463582fc436cd0ce0f10a6f4610771cfa7c tools/xmlGenerator/domainGeneratorConnector.cpp 97'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/322#discussion_r48042874'>File: tools/xmlGenerator/domainGeneratorConnector.cpp:L1-205</a></b>
- <a href='https://github.com/dawagner'></a> Missing closing `'` before ` rejected`.
- [x] <a href='#crh-comment-Pull 0b584463582fc436cd0ce0f10a6f4610771cfa7c newDomainGenerator.py 1'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/322#discussion_r48021954'>File: newDomainGenerator.py:L1-324</a></b>
- <a href='https://github.com/dawagner'></a> oops, I did not intend to add this file. I'll remove it.
- [x] <a href='#crh-comment-Pull 0b584463582fc436cd0ce0f10a6f4610771cfa7c tools/xmlGenerator/domainGenerator.py 223'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/322#discussion_r48022071'>File: tools/xmlGenerator/domainGenerator.py:L159-208</a></b>
- <a href='https://github.com/dawagner'></a> no need to flush here.
- <a href='https://github.com/krocard'></a> done
- [x] <a href='#crh-comment-Pull 0b584463582fc436cd0ce0f10a6f4610771cfa7c tools/xmlGenerator/domainGenerator.py 246'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/322#discussion_r48022117'>File: tools/xmlGenerator/domainGenerator.py:L159-208</a></b>
- <a href='https://github.com/dawagner'></a> flush may not be needed.
- [x] <a href='#crh-comment-Pull 0b584463582fc436cd0ce0f10a6f4610771cfa7c tools/xmlGenerator/domainGeneratorConnector.cpp 29'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/322#discussion_r48022312'>File: tools/xmlGenerator/domainGeneratorConnector.cpp:L1-205</a></b>
- <a href='https://github.com/dawagner'></a> Maybe we could unconditionally instantiate the logger: it doesn't cost much.
- <a href='https://github.com/krocard'></a> I think it is just as good as is. No need to complexity the code.
- [x] <a href='#crh-comment-Pull 0b584463582fc436cd0ce0f10a6f4610771cfa7c tools/xmlGenerator/domainGeneratorConnector.cpp 138'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/322#discussion_r48043053'>File: tools/xmlGenerator/domainGeneratorConnector.cpp:L1-205</a></b>
- <a href='https://github.com/dawagner'></a> Throw an error.
- <a href='https://github.com/krocard'></a> :+1: The code now continues on error but keep track of how many occurred and report it with the exit code.
- [x] <a href='#crh-comment-Pull 0b584463582fc436cd0ce0f10a6f4610771cfa7c tools/xmlGenerator/domainGenerator.py 248'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/322#discussion_r48043204'>File: tools/xmlGenerator/domainGenerator.py:L159-208</a></b>
- <a href='https://github.com/dawagner'></a> Forward the connector's return code.

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/322?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/322?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/322'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: newDomainGenerator.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
+#! /usr/bin/python
```
Comment (dawagner) [inline newDomainGenerator.py]: oops, I did not intend to add this file. I'll remove it.

File: tools/xmlGenerator/domainGenerator.py [hunk 2]
```diff
@@ -238,88 +159,49 @@ def info(self, message):
...
-    ok, domains, error = pfw.exportDomainsXml("", True, False)
-    sys.stdout.write(domains)
+            instruction = "createSelectionCriterion\0{}\0{}\0{}".format(
+                    criterion['inclusive'],
+                    criterion['name'],
+                    "\0".join(criterion['values']))
+            connector.stdin.write("{}\n".format(instruction))
+
+        connector.stdin.write("start\n")
+        connector.stdin.flush()
```
Comment (dawagner) [inline tools/xmlGenerator/domainGenerator.py]: no need to flush here.

File: tools/xmlGenerator/domainGenerator.py [hunk 3]
```diff
@@ -238,88 +159,49 @@ def info(self, message):
...
+        # Generate the script for each EDD file
+        for filename, parsed_edd in parsed_edds:
+            logging.info("Translating and injecting EDD file {}".format(filename))
+            translator = PfwScriptTranslator(separator="\0")
+            parsed_edd.translate(translator)
+            connector.stdin.write("\n".join(translator.getScript()))
+            connector.stdin.write("\n")
+
+        # Closing the connector's input triggers the domain generation
+        connector.stdin.flush()
```
Comment (dawagner) [inline tools/xmlGenerator/domainGenerator.py]: flush may not be needed.

File: tools/xmlGenerator/domainGeneratorConnector.cpp [hunk 4]
```diff
@@ -0,0 +1,204 @@
+#include <ParameterMgrFullConnector.h>
```
Comment (dawagner) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: missing license header

File: tools/xmlGenerator/domainGeneratorConnector.cpp [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+#include <Tokenizer.h>
+
+#include <iostream>
+#include <sstream>
+#include <memory>
+#include <string>
+#include <stdexcept>
+
+using std::string;
+using std::runtime_error;
```
Comment (dawagner) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: Add a custom exception class to differentiate Parameter Framework errors from other ones.

File: tools/xmlGenerator/domainGeneratorConnector.cpp [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+
+class XmlGenerator
+{
+public:
+    XmlGenerator(const string &toplevelConfig, bool validate, bool verbose, string schemasDir)
+        : mConnector(toplevelConfig),
+          mCommandHandler(mConnector.createCommandHandler())
+    {
+        if (verbose) {
+            mLogger.reset(new MyLogger);
```
Comment (dawagner) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: Maybe we could unconditionally instantiate the logger: it doesn't cost much.

File: tools/xmlGenerator/domainGeneratorConnector.cpp [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+        // Disable irrelevant failure conditions
+        mConnector.setFailureOnMissingSubsystem(false);
+        mConnector.setFailureOnFailedSettingsLoad(false);
+
+        // Disable the remote interface because we don't need it and it might
+        // get in the way (e.g. the port is already in use)
+        mConnector.setForceNoRemoteInterface(true);
+    }
+
+    /** Read each line of the input stream and takes an action accordingly
```
Comment (dawagner) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: Reads

File: tools/xmlGenerator/domainGeneratorConnector.cpp [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+    if (criterionType == nullptr) {
+        throw runtime_error("Failed to create an " +
+                        string(inclusiveness ? "inclusive" : "exclusive") +
+                        " criterion type");
+    }
+
+    int index = 0;
+    for (const auto &literalValue : tokens) {
+        // inclusive criteria are bitfields
+        int numericalCriterionValue = inclusiveness ? 1 << index : index;
```
Comment (dawagner) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: `numericalValue` (for naming consistency)

File: tools/xmlGenerator/domainGeneratorConnector.cpp [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+        string error;
+        bool success = criterionType->addValuePair(numericalCriterionValue, literalValue, error);
+
+        if (not success) {
+            std::ostringstream message;
+            message << "Valuepair '" << numericalCriterionValue << "/" << literalValue
+                    << " rejected for " << name << ". (" << error << ")";
```
Comment (dawagner) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: Missing closing `'` before ` rejected`.

File: tools/xmlGenerator/domainGeneratorConnector.cpp [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+        tokens.erase(begin(tokens)); // drop the command name
+
+        if (command == "createSelectionCriterion") {
+            addCriteria(tokens);
+        } else if (command == "start") {
+            start();
+        } else {
+            string output;
+            if (not mCommandHandler->process(command, tokens, output)) {
+                std::cerr << output << std::endl;
```
Comment (dawagner) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: Throw an error.

File: tools/xmlGenerator/domainGenerator.py [hunk 11]
```diff
@@ -238,88 +159,49 @@ def info(self, message):
...
+        connector.stdin.close()
+        connector.wait()
```
Comment (dawagner) [inline tools/xmlGenerator/domainGenerator.py]: Forward the connector's return code.

File: tools/xmlGenerator/domainGenerator.py (hunk 2 above)
Comment (krocard) [inline tools/xmlGenerator/domainGenerator.py]: done

File: tools/xmlGenerator/domainGenerator.py (hunk 3 above)
Comment (krocard) [inline tools/xmlGenerator/domainGenerator.py]: :+1:

File: tools/xmlGenerator/domainGeneratorConnector.cpp (hunk 6 above)
Comment (krocard) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: I think it is just as good as is. No need to complexity the code.

Comment (krocard) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: :+1:

File: tools/xmlGenerator/domainGeneratorConnector.cpp (hunk 10 above)
Comment (krocard) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: :+1: The code now continues on error but keep track of how many occurred and report it with the exit code.

File: tools/xmlGenerator/domainGenerator.py (hunk 11 above)
Comment (krocard) [inline tools/xmlGenerator/domainGenerator.py]: :+1:

File: tools/xmlGenerator/domainGenerator.py (hunk 2 above)
Comment (krocard) [inline tools/xmlGenerator/domainGenerator.py]: :+1:

File: newDomainGenerator.py (hunk 1 above)
Comment (krocard) [inline newDomainGenerator.py]: :+1:

**Comment by krocard** [comment]
Taken over by #324. :+1:

File: tools/xmlGenerator/domainGeneratorConnector.cpp (hunk 7 above)
Comment (krocard) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: :+1:<a href='#crh-update-none'></a>

File: tools/xmlGenerator/domainGeneratorConnector.cpp (hunk 9 above)
Comment (krocard) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: :+1:<a href='#crh-update-none'></a>

File: tools/xmlGenerator/domainGeneratorConnector.cpp (hunk 8 above)
Comment (krocard) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: :+1:<a href='#crh-update-none'></a>

File: tools/xmlGenerator/domainGeneratorConnector.cpp (hunk 5 above)
Comment (krocard) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: :+1:<a href='#crh-update-none'></a>

File: tools/xmlGenerator/domainGeneratorConnector.cpp (hunk 4 above)
Comment (krocard) [inline tools/xmlGenerator/domainGeneratorConnector.cpp]: :+1:<a href='#crh-update-complete'></a>

**Pull request closed without merging**